framework/src/tinyalsa/tinyalsa.c

```diff
@@ -311,6 +308,16 @@ int pcm_set_config(struct pcm *pcm, const struct pcm_config *config)
 		pcm->config = *config;
 	}
 
+	if (ioctl(pcm->fd, AUDIOIOC_GETBUFFERINFO, (unsigned long)&buf_info) < 0) {
```

Comment:
This ioctl must be moved below after AUDIOIOC_CONFIGURE so that first the sample rate is set in driver and later based on the sample rate we will get buffer size.

Reply:
yeah you're right. I will fix